src/Nest/Mapping/DynamicMapping.cs

```diff
@@ -13,4 +15,60 @@ public enum DynamicMapping
 		[EnumMember(Value = "strict")]
 		Strict
 	}
+
+	internal class DynamicMappingFormatter : IJsonFormatter<Union<bool,DynamicMapping>>
```

Comment:
Yet another formatter :) 

Easier said then done but I wonder if we can formulate other strategies around this.

Reply:
An alternative would be to have a more lenient `BoolFormatter` that can deserialize from string values. My initial resistance in going this direction is that it is a change that affects the performance of the general case of deserializing boolean values in order to handle the less frequent case of deserializing sting bool values.

Reply:
Was talking more about making writing formatters easier. 

The generalized bool formatter might be a better solution functionality since even if we never write strings for booleans Elasticsearch will happily ingest `"true"` and return as is from _source. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/boolean.html

So data that came in from other places might be stringified books too.

Reply:
that's a good point. Perhaps we need to be more lenient on numeric types too?